File: java/mirror.java
import java.util.Scanner;

public class mirror {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int T = scan.nextInt();
		for (int i = 1; i <= T; i++) {
			int R = scan.nextInt();
			int C = scan.nextInt();
			char[][] image = new char[R][C];
			for (int j = 0; j < R; j++) {
				String line = scan.next();
				for (int k = 0; k < C; k++)
					image[j][k] = line.charAt(k);
			}
			System.out.printf("Test %d%n", i);
			for (int j = R - 1; j >= 0; j--) {
				for (int k = C - 1; k >= 0; k--)
					System.out.print(image[j][k]);
				System.out.println();
			}
		}
		scan.close();
	}
}

File: ts/src/mirror.ts
import readline = require('readline')

function reverseImage (image: string[][]): string[][] {
  return image.reverse().map(row => row.reverse())
}

function printImage (image: string[][]): void {
  image.forEach(row => console.log(row.join('')))
}

let lineCount = 0
let height = 0
let image: string[][] = []
let cases = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      if (height === 0) {
        const [row] = line.split(' ').map(Number)
        height = row ?? 0
        image = []
        cases++
      } else {
        image.push(line.split('') as string[])
        height--
        if (height === 0) {
          console.log(`Test ${cases}`)
          printImage(reverseImage(image))
        }
      }
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
